### PR TITLE
[PLATFORM-563] Module preview graphics

### PR DESCRIPTION
**Pull request by mondoreale**
Table

May-05-2019 00-17-57

Chart

May-05-2019 00-19-59

### CODE REVIEW

**Comment by mondoreale** [comment]
There's also one for Comment module but haven't made a screenshot of it.

File: app/src/editor/canvas/components/Preview.jsx [hunk 1]
```diff
@@ -19,144 +120,130 @@ function getModuleKey(m) {
...
         key: getModuleKey(m),
         top: Number.parseInt(m.layout.position.top, 10) || 0,
         left: Number.parseInt(m.layout.position.left, 10) || 0,
         height: Number.parseInt(m.layout.height, 10) || defaultLayout.height,
         width: Number.parseInt(m.layout.width, 10) || defaultLayout.width,
-        isResizable: isModuleResizable(m),
-        portRows: getPortRows(m),
         title: (m.displayName || m.name),
-        type: module.widget || m.jsModule,
+        type: (m.uiChannel && m.uiChannel.webcomponent) || m.widgth || m.jsModule,
```
Comment (timoxley) [inline app/src/editor/canvas/components/Preview.jsx]: widgth?

File: app/src/editor/dashboard/components/Preview.jsx [hunk 2]
```diff
@@ -66,10 +72,11 @@ export default function Preview({
                         >
                             <g transform="scale(0.95)">
                                 <ModulePreview
-                                    um={0.25}
+                                    um={0.2}
```
Comment (timoxley) [inline app/src/editor/dashboard/components/Preview.jsx]: This can go now

File: app/src/editor/canvas/components/Cables.jsx [hunk 3]
```diff
@@ -30,6 +30,10 @@ export function Cable({ cable }) {
                 to.left,
                 to.top,
             )}
+            stroke="#525252"
+            fill="none"
+            strokeWidth="1"
```
Comment (timoxley) [inline app/src/editor/canvas/components/Cables.jsx]: Why move from css?

File: app/src/editor/canvas/components/Preview.jsx (hunk 1 above)
Comment (mondoreale) [reply app/src/editor/canvas/components/Preview.jsx]: A typo, omg. 🔨

File: app/src/editor/dashboard/components/Preview.jsx (hunk 2 above)
Comment (mondoreale) [reply app/src/editor/dashboard/components/Preview.jsx]: It sure can! 👌

File: app/src/editor/canvas/components/Cables.jsx (hunk 3 above)
Comment (mondoreale) [reply app/src/editor/canvas/components/Cables.jsx]: Apparently attributes have lower prio than styling, so in order to customize strokes or fills I'd have to define another class.

Comment (timoxley) [reply app/src/editor/canvas/components/Cables.jsx]: 👍

Comment (timoxley) [reply app/src/editor/canvas/components/Cables.jsx]: you could maybe apply the customisation as an inline style property

**Comment by mattatgit** [comment]
@mondoreale @timoxley Can we make the white-ish modules completely FFFFFF please ? The non-FFFFFF colour is a mistake of ours still hanging around, sorry about that! :)

**Comment by mondoreale** [comment]
@mattatgit @timoxley you guys are gonna love it! (altho love is never certain) ❤️ 

There you go!

Canvases (not many changes here since the last update):

Screenshot 2019-05-21 12 02 00

And them dashboards, man:

Screenshot 2019-05-21 12 03 10

@timoxley I moved a lot of code to `$editor/shared`. It's kinda cool; DRY eh!?

**Comment by mondoreale** [comment]
Dashboard modules have titles and types so gray bars that represent titles have dynamic widths. Types let us display tables, comments and charts properly.

**Comment by mattatgit** [comment]
@mondoreale Looks great, nice work sir!

**Comment by mondoreale** [comment]
@timoxley, if you have a second, throw an eye at this one more time, please. I'm gonna merge it tomorrow morning CET. 💣

**Pull request merged**